luaui/Widgets/cmd_quick_build_extractor.lua

```diff
@@ -0,0 +1,263 @@
+function widget:GetInfo()
+	return {
+		name = "Quick Build (mex/geo)",
+		desc = "Adds ability to quickly place or upgrade mex/geothermal by right clicking.",
+		author = "Hobo Joe, based on work by Google Frog, NTG, Chojin, Doo, Floris, and Tarte",
+		version = "1.0",
+		date = "Jan 2024",
+		license = "GNU GPL, v2 or later",
+		handler = true,
+		layer = 1000,
+		enabled = true
+	}
+end
+
+local CMD_MOVE = CMD.MOVE
+local CMD_GUARD = CMD.GUARD
+local CMD_RECLAIM = CMD.RECLAIM
+
+local spGetActiveCommand = Spring.GetActiveCommand
+local spGetUnitDefID = Spring.GetUnitDefID
+local spGetUnitPosition = Spring.GetUnitPosition
+
+local geoPlacementRadius = 5000	-- (not actual ingame distance)
+local mexPlacementRadius = 2000	-- (not actual ingame distance)
```

Comment:
why 5k and 2k? why not 4k and 3k.

Reply:
Arbitrary, basically what "feels" right. I'll write a bigger comment with more explanation about why other things like spotsize aren't used.